src/components/molecules/create-survey/new-survey-header.tsx

```diff
@@ -1,12 +1,133 @@
+import { useFormContext } from "react-hook-form";
+import dayjs from "dayjs";
+import EditButton from "@/components/atoms/new-survey/edit-button";
+import SaveButton from "@/components/atoms/new-survey/save-button";
+import TextFormField from "@/components/atoms/form/fields/text-form-field";
 
 const NewSurveyHeader = () => {
-  // TODO: implement
-  // zmiana stanu isEditing i odpowiednie przyciski
-  // pola tekstowe na nazwę ankiety i opis
-  // zwyczajny widok bez edycji
+  const [isEditing, setIsEditing] = useState(false);
+  const { getValues, register } = useFormContext();
+
+  const [title, setTitle] = useState("Nowa ankieta");
+  const [description, setDescription] = useState("Proszę wypełnić ankietę");
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+
+  const handleEditClick = () => {
+    setIsEditing(true);
+  };
+
+  const handleSaveClick = () => {
+    setTitle(getValues("surveyTitle") ?? "Moja ankieta");
+    setDescription(getValues("surveyDescription") ?? "");
+    setStartDate(dayjs(getValues("surveyStartDate")).format("YYYY-MM-DD"));
+    setEndDate(dayjs(getValues("surveyEndDate")).format("YYYY-MM-DD"));
+    /*
+    console.log("Title:", title);
+    console.log("Description:", description);
+    console.log("Start Date:", startDate);
+    console.log("Start Date:", endDate);
+*/
+    setIsEditing(false);
+  };
+
   return (
-    <SurveyHeader title="New Survey" description="Please fill out the survey" />
+    <div>
+      {isEditing ? (
+        <>
+          <div className="mb-4 w-full rounded-lg bg-white p-6 text-left shadow">
+            <label
+              htmlFor="surveyTitle"
+              className="block text-sm font-bold text-gray-600"
+            >
+              Podaj tytuł ankiety
+            </label>
+            <TextFormField
+              name="surveyTitle"
+              type="text"
+              className="w-full rounded-md border px-3 py-2 focus:border-blue-500 focus:outline-none"
+              required
+            />
+          </div>
+          <div className="mb-4 w-full rounded-lg bg-white p-6 text-left shadow">
+            <label
+              htmlFor="surveyDescription"
+              className="block text-sm font-bold text-gray-600"
+            >
+              Podaj opis ankiety
+            </label>
+            <TextFormField
+              name="surveyDescription"
+              type="text"
+              className="w-full rounded-md border px-3 py-2 focus:border-blue-500 focus:outline-none"
+            />
+          </div>
+          <div className="mb-4 w-full rounded-lg bg-white p-6 text-left shadow">
+            <label
+              htmlFor="surveyStartDate"
+              className="block text-sm font-bold text-gray-600"
+            >
+              Podaj datę rozpoczęcia ankiety
+            </label>
+            <TextFormField
+              name="surveyStartDate"
+              type="date"
```

Comment:
jeśli to działa to trzeba do TextFormField dodać do typu "date" żeby się eslint nie czepiał